packages/Webkul/Shop/src/Database/Migrations/2019_11_21_194608_add_stored_function_to_get_url_path_of_category.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+use Illuminate\Support\Facades\DB;
+use Illuminate\Database\Migrations\Migration;
+
+class AddStoredFunctionToGetUrlPathOfCategory extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        $functionSQL = <<< SQL
+            DROP FUNCTION IF EXISTS `get_url_path_of_category`;
+            CREATE FUNCTION get_url_path_of_category(
+                categoryId INT,
+                localeCode VARCHAR(255)
+            )
+            RETURNS VARCHAR(255)
+            DETERMINISTIC
+            BEGIN
+            
+                DECLARE urlPath VARCHAR(255);
+                -- Category with id 1 is root by default
+                IF categoryId <> 1
```

Comment:
This line will not work for shops who hadn't truncated their DBs before going to production after their development.

And SAAS based scenario have multiple root categories and it will only check for id != 1, while SAAS + Bagisto can have multiple root category ids.

Reply:
Hi @popperhead 

Your concern is absolutely right, it never gonna work on SAAS due to multiple root categories.
@hb-monah  @popperhead  can we use levels of categories to avoid this one.

Reply:
Hi @popperhead @rahulshukla-webkul 

thank you. 
I will provide a solution for multiple root categories soon.

Reply:
@hb-monah @rahulshukla-webkul I don't think you should provide solution for this. The one concerned with SAAS development has to add in the module.

Reply:
Well, as Bagisto provides the possibility to use multiple root categories there should be a solution for this (already in the making, needs testing though).
@popperhead  @rahulshukla-webkul